youtube_dl/extractor/hketv.py

```diff
@@ -0,0 +1,157 @@
+            'upload_date': '20181024',
+            'duration': 900,
+            'subtitles': {
+                'en': [{
+                    'url': 'https://apps.hkedcity.net/media/mediaplayer/caption.php?f=74395&lang=en',
+                    'ext': 'srt',
+                }],
+                'zh-Hant': [{
+                    'url': 'https://apps.hkedcity.net/media/mediaplayer/caption.php?f=74395&lang=qmt',
+                    'ext': 'srt',
+                }],
+            }
+        },
+    }]
+
+    _CC_LANGS = {
+        '中文（繁體中文）': 'zh-Hant',
+        '中文（简体中文）': 'zh-Hans',
+        'English': 'en',
+        'Bahasa Indonesia': 'id',
+        '\u0939\u093f\u0928\u094d\u0926\u0940': 'hi',
+        '\u0928\u0947\u092a\u093e\u0932\u0940': 'ne',
+        'Tagalog': 'tl',
+        '\u0e44\u0e17\u0e22': 'th',
+        '\u0627\u0631\u062f\u0648': 'ur',
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_meta('ed_title', webpage, fatal=True)
+
+        file_id = self._html_search_regex(r'post_var\["file_id"\]\s*=\s*(.+?);', webpage, 'file ID')
+        curr_url = self._html_search_regex(r'post_var\["curr_url"\]\s*=\s*"(.+?)";', webpage, 'curr URL')
+        data = {
+            'action': 'get_info',
+            'curr_url': curr_url,
+            'file_id': file_id,
+            'video_url': file_id,
+        }
+        _APPS_BASE_URL = 'https://apps.hkedcity.net'
+        handler_url = _APPS_BASE_URL + '/media/play/handler.php'
+
+        response = self._download_json(
+            handler_url, video_id,
+            data=urlencode_postdata(data),
+            headers=merge_dicts({'Content-Type': 'application/x-www-form-urlencoded'},
+                                self.geo_verification_headers()))
+
+        result = response.get('result')
+
+        formats = []
+        subtitles = {}
+
+        if response.get('success') and response.get('access'):
+            width = str_to_int(result.get('width'))
+            height = str_to_int(result.get('height'))
+
+            playlist0 = try_get(result, lambda x: x['playlist'][0], dict)
+            fmts = try_get(playlist0, lambda x: x['sources'], list)
+            for fmt in fmts:
+                label = fmt.get('label')
+                if label == 'HD':
+                    h = 720
+                elif label == 'SD':
+                    h = 360
+                w = h * width // height
+                urlh = self._downloader.urlopen(_APPS_BASE_URL + fmt.get('file'))
+                formats.append({
+                    'format_id': label,
+                    'ext': fmt.get('type'),
+                    'url': urlh.geturl(),
+                    'width': w,
+                    'height': h,
+                })
+
+            tracks = try_get(playlist0, lambda x: x['tracks'], list)
+            for track in tracks:
```

Comment:
`tracks` is not guaranteed to be iterable.

Reply:
Thanks!  Will be changing tracks to this instead:
```
            tracks = playlist0.get('tracks', [])
```